Code/Framework/AzCore/AzCore/EBus/Environment.h

```diff
@@ -250,7 +250,8 @@ namespace AZ
     template<class Context>
     Context* EBusEnvironmentStoragePolicy<Context>::Get()
     {
-        if (!s_defaultGlobalContext && Environment::IsReady())
+        if ((!s_defaultGlobalContext || (s_defaultGlobalContext && !s_defaultGlobalContext.IsConstructed()))
```

Comment:
I filed this issue with the Lumberyard issue system for Lumberyard 1.x.
We(@carlitosan) have seen this issue with the ScriptCanvas test when running with a gtest_repeat count >1.

The biggest issue is the caching of the EBus Context itself.
When the dll that is the owner of the EBus context(The first dll that calls BusConnect/BusDisconnect) unloads, the other dlls don't know that the "owner dll" hass unloaded, so it keeps a pointer to the old EnvironmentVariable instance.

When the owner dll reloads, a new instance of the EBusContext Environment variable is created in the owner dll memory space, but as you have investigated, the other dlls copies of `s_defaultGlobalContext` passes the `operator ` bool` check because the [EnvironmentVariableHolder](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Module/Environment.h#L479-L482) m_data instance is non-nullptr, but the [constructed state ](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Module/Environment.h#L489-L492) is false.

You can actually simplify this check to just check the `!IsConstructed()` state, since that also encompasses the IsValid check

```suggestion
        if (!s_defaultGlobalContext.IsConstructed()
```

Reply:
Also one more thing to note.
On MacOS, we have seen that the dylibs that are loaded aren't unloaded, so an issue occurs when static deinitialization and inititalization will not re-run for a shared library, because it was never unloaded.

You may want to run these the googletest for test shared library using the `--gtest_repeat=2` option and see if it passes on Windows and MacOS